components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -54,9 +55,18 @@ base::Value::List BuildMessages(
   {
     std::string date_and_time_string =
         base::UTF16ToUTF8(TimeFormatFriendlyDateAndTime(base::Time::Now()));
-    std::string system_message = base::StrCat({base::ReplaceStringPlaceholders(
-        l10n_util::GetStringUTF8(IDS_AI_CHAT_LLAMA2_SYSTEM_MESSAGE_GENERIC),
-        {date_and_time_string}, nullptr)});
+    std::string system_message;
+
+    if (!model_options.model_system_prompt.empty()) {
```

Comment:
I wonder if a `std::optional` makes more sense here?

Reply:
@fallaciousreasoning Can you elaborate? The `system_message` _will be_ populated, either in the first condition block (user-defined prompt) or by the else-block (default prompt).